plugins/MobileMessaging/Controller.php

```diff
@@ -155,4 +158,32 @@ public function getCredentialFields()
 
         return $view->render();
     }
+
+    public function getBrowserNotifications()
+    {
+        $login = null;
+        $token = Common::getRequestVar('token');
+
+        if ($token) {
+            $user = (new Model())->getUserByNotificationToken($token);
+            if ($user) {
+                $login = $user['login'];
+            }
+        }
+
+        if (! $login) {
+            throw new NoAccessException('Missing or invalid token');
+        }
+
+        $optionKey = 'ScheduledReports.notifications.' . $login;
```

Comment:
This prefix `ScheduledReports.notifications.'` is used a few times, be good to make it a constant.